Geometry_Engine/Query/IsNull.cs

```diff
@@ -0,0 +1,64 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Geometry;
+using BH.oM.Geometry;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Quantities.Attributes;
+using System.ComponentModel;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
+
+namespace BH.Engine.Geometry
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Checks if an Geometry is null and outputs relevant error message.")]
+        [Input("geometry", "The Geometry to test for null.")]
+        [Input("methodName", "The name of the method to reference in the error message.")]
+        [Input("msg", "Optional message to be returned in addition to the generated error message.")]
+        [Output("isNull", "True if the Geometry is null.")]
+        public static bool IsNull(this IGeometry geometry, [CallerMemberName] string methodName = "", string msg = "")
```

Comment:
Probably one for later, to not bloat this PR as it is supposed to handle structures_Engine, but raising now to potentially avoid refactoring later.

Was thinking if we wanted to have this method look deeper, as in, if for example a line is first checked if it is null itself, then have both of its points checked. This could be controlled via some boolean input on the method like `deepCheck` or similar, and then would need a specific method for each type.

again, not for this PR to implement, but raising jsut in case we want to add the bool straight away.

Reply:
Add to the Geometry_Engine null check handling issue here: https://github.com/BHoM/BHoM_Engine/issues/2487